### ISSUE TITLE
CompleteActivityByID return "activity task not found" error, even the activity should be existed

### ISSUE THREAD

**Issue by enuoCM** [uber/cadence#3334]
**Describe the bug**
If call CompleteActivityByID with wrong ActivityID, then call it again with right ActivityID, it will always return "activity task not found" error, even the activity should be existed.

cadense-client:0.12.1
cadense server: ubercadence/server:master-auto-setup

**To Reproduce**
Is the issue reproducible? 
Yes

**Steps to reproduce the behavior:**
A workflow with activity of type A. 
Workflow options:
```
executionStartToCloseTimeout: 10m
taskStartToCloseTimeout: 1m
```
Activity options:
```
scheduleToStartTimeout: 1m
startToCloseTimeout: 5m
```

1. Activity A returns activity.ErrResultPending and assume it's ActivityID is 1.
2. Call CompleteActivityByID with right **domain**, **Workflow id**, **Run id** but wrong **activity id**, such as 100. It returns "**Cannot locate Activity ScheduleID**" error, which is expected.
3. Call CompleteActivityByID again with right **domain**, **Workflow id**, **Run id** and **activity id** (Here is 1). It returns "**activity task not found**" error, which is not expected. And then the Activity A can not **CompleteActivityByID** and **CompleteActivity**(by token) any more.

If step2 and step3(or do not execute step3) is executed 1m (scheduleToStartTimeout of Activity A) **later** after Activity A started, this workflow will continue with ActivityTaskStarted status and failed until 10m(executionStartToCloseTimeout of the workflow) later.

If step2 and step 3(or do no execute step3) is executed **within** 1m (scheduleToStartTimeout of Activity A) after Activity A started, Activity A will failed with "ActivityTaskTimedOut, with detail=[], startedEventId=-23, timeoutType=SCHEDULE_TO_START info". And the whole workflow will be failed in 1m with "cadenceInternal:Timeout SCHEDULE_TO_START" reason
  
**Expected behavior**
Step2 throw error and step3 should be successfully executed.
No matter to call **CompleteActivityByID** or **CompleteActivity** repeatedly with the wrong or right params, it should not affect the next right invocation of **CompleteActivityByID** or **CompleteActivity**.

**Comment by enuoCM** [comment]
Hi, does anybody have the same problem? It's easy to reproduce really.

**Comment by enuoCM** [comment]
execute `workflowClient.DescribeWorkflowExecution(context.Background(), "workflowID-A", "runID-A")` got wer.PendingActivities which contains a PendingActivity with ActivityID = "6". 
continue execute `workflowClient.CompleteActivityByID(context.Background(), domain, "workflowID-A", "runID-A", "", result, nil)`, it returns EntityNotExistsError. Please help. Thanks.

**Comment by meiliang86** [comment]
CompleteActivityByID only works when your activity has been started, but not timed out yet. Can you check your workflow history to see if that is the case? Also history should tell you which ID is assigned for the activity.

**Comment by enuoCM** [comment]
@meiliang86 the activity is started and not timed out.
@chris-rcn postgres

**Comment by chris-rcn** [comment]
I had a similar problem and found a fix:
https://github.com/uber/cadence/issues/3482

**Comment by enuoCM** [comment]
@chris-rcn Great! Can this issue be reproduced after your fix? Thanks.

**Comment by chris-rcn** [comment]
The issue that I was facing is not reproducible after the change (otherwise, it wouldn't be much of a fix).

My guess (based on the severity of the bug) is that it will fix lots of strange Postgres bugs.

**Comment by enuoCM** [comment]
@chris-rcn Thanks. I verified that this issue is fixed by your commit too. I'll close this issue after new version released.

**Comment by chris-rcn** [comment]
Please note that the fix is actually different from what I originally proposed.  This is the correct fix:  https://github.com/uber/cadence/pull/3498

**Comment by enuoCM** [comment]
Closed as already released 0.14.2
